### ISSUE TITLE
Localhost deployment fails to detect WAN IP on Aliyun 

### ISSUE THREAD

**Issue by nickolasclarke** [jlund/streisand#889]
### Expected behavior:
`streisand_ipv4_address` should be set to the external IP
### Actual Behavior:
`streisand_ipv4_address` gets set to instance's internal IP, which causes most services to break.
### Steps to Reproduce:
1.

**Comment by nickolasclarke** [comment]
currently we try our darnedest to figure out what the external IP is in the localhost.yml and detect-provider.yml playbooks. We have workarounds for AWS and GCE. Aliyun is similar to those platforms. Since Aliyun is not currently a supported platform, and this could possibly be an issue on other providers, I wonder if it would be best to simply prompt for an external IP when the localhost deployment is selected, and try our best otherwise if that is not filled.

thoughts? Is there a more elegant way to get at a WAN IP that would be more universal than what we have already?

**Comment by cpu** [comment]
Could you just set `streisand_ipv4_address` as an inventory host var?

**Comment by nickolasclarke** [comment]
@cpu yes, I could. Short of manually editing the file, I'd like to add a conditional prompt for that. At first I thought this should be an env var that is consumed by a playbook later to set `streisand_ipv4_address`, but then I thought it may be more appropriate to add a `vars_prompt` to playbooks/localhost.yml. 

In this I'll:
1. prompt for the user to set `streisand_ipv4_address` if they want to. This would go above the `tasks` block. I'd need to make it conditional in case the `provider-detect.yml` successfully set `streisand_ipv4_address` Otherwise, it should prompt.
2.  then make the tasks block conditional that tries to set` streisand_ipv4_address` with `ansible_default_ipv4.address` if the above approaches do not set the var. 

Does that approach make sense?

**Comment by cpu** [comment]
It seems like there are all sorts of problems here unique to Aliyun. I'm still expressing an opinion that we should be extremely selective in what new code we add to Streisand and another provider is a big commitment, especially if it appears to be all sorts of non-standard. 

> change ownership to root:nogroup

The [shadowsocks role already does this](https://github.com/jlund/streisand/blob/89754934b1813cc2870f7bae24431988a0c2ff03/playbooks/roles/shadowsocks/tasks/main.yml#L102:L104). Can you determine why it didn't work on Aliyun but works elsewhere?

**Comment by nickolasclarke** [comment]
@cpu yeah, I agree. I'm not interested in making any code changes that are specific to aliyun, but rather make changes that would potentially apply to all non-standard deployments.

**Comment by cpu** [comment]
Makes sense! Improvements in that department would be great. Thanks for clarifying!

**Comment by hadifarnoud** [comment]
I did manage to install streisand via 'existing server' option. it worked just fine.

would be great to have specific support with 'security group' configuration and all that.

**Comment by nickolasclarke** [comment]
@hadifarnoud I'm working on this on a branch off of my fork. Its looking pretty straightforward so far, so hopefully we'll have it soon!
